pkg/endpoint/policy.go

```diff
@@ -481,6 +482,19 @@ func (e *Endpoint) Regenerate(regenMetadata *regeneration.ExternalRegenerationMe
 
 		var buildSuccess bool
 		var regenError error
+		defer func() {
```

Comment:
I'm not sure I follow why this became a defer. Unless we're crashing in this function it wouldn't make a difference, right? (this isn't a blocker, just a question)

Reply:
The code went through a lot of changes before it got to this final state, which needed this `defer`. It doesn't any more, so I'll put it at the end of the goroutine.